vnext/local-cli/runWindows/utils/msbuildtools.js

```diff
@@ -61,27 +62,37 @@ class MSBuildTools {
       newInfo(`Build target: ${target}`);
     }
     const verbosityOption = verbose ? 'normal' : 'minimal';
-    const errorLog = path.join(
-      process.env.temp,
-      `msbuild_${process.pid}${target ? '_' + target : ''}.err`,
+    const logPrefix = path.join(
+      buildLogDirectory || process.env.temp,
+      `msbuild_${process.pid}${target ? '_' + target : ''}`,
     );
+
+    const errorLog = logPrefix + '.err';
+    const warnLog = logPrefix + '.wrn';
+    const binlog = buildLogDirectory
+      ? ':' + logPrefix + '.binlog'
+      : target
+        ? `:${target}.binlog`
+        : '';
```

Comment:
Could we avoid the nested ternaries?  This is a bit hard to reason about as-is.

Reply:
I'll send a separate Pr for that after this one, to keep the changes small. Note the file is not TS, nor flow so there will not be any checking for args.